packages/service/ni_measurement_plugin_sdk_service/_internal/parameter/serialization_descriptors.py

```diff
@@ -1,76 +1,43 @@
-    file_descriptor: descriptor_pb2.FileDescriptorProto,
-) -> Any:
-    """Get's matching enum class from 'file_descriptor'.
 
-    Args:
-        metadata_enum_list (List[str]): Enum names from metadata.annotations.
 
-        file_descriptor: Descriptor of proto file.
-
-    Returns:
-        Any: Matching enum class in a str type or None when enum is protobuf.
-    """
-    for enum_type in file_descriptor.enum_type:
-        enum_names = [enum_value.name for enum_value in enum_type.value]
-        if sorted(metadata_enum_list) == sorted(enum_names):
-            return enum_type.name
-    return None
-
-
-def _create_enum_type(
+def _create_enum_type_class(
     file_descriptor: descriptor_pb2.FileDescriptorProto,
     parameter_metadata: ParameterMetadata,
     field_descriptor: FieldDescriptorProto,
 ) -> None:
-    """Implement a enum class in 'file_descriptor'.
-
-    Args:
-        file_descriptor (FileDescriptorProto): Descriptor of a proto file.
-
-        parmeter_metadata (ParameterMetadata): Metadata of current field.
-
-        field_descriptor (FieldDescriptorProto): Descriptor of a field.
-
-    Returns:
-        None: Only creates a enum class in file_descriptor.
-    """
+    """Implement a enum class in 'file_descriptor'."""
     enum_dict = loads(parameter_metadata.annotations[ENUM_VALUES_KEY])
-    if parameter_metadata.default_value is None:
-        enum_type_name = _get_output_enum_type(
-            metadata_enum_list=[enum for enum in enum_dict.keys()], file_descriptor=file_descriptor
-        )
-    else:
-        enum_type_name = _get_enum_type(parameter_metadata).__name__
+    enum_type_name = _get_enum_type(parameter_metadata).__name__
+
+    # if enum is a protobuf then enum_type_name is 1st letter of each enum name
+    # e.g. {"NONE": 0, "RED": 1, "GREEN": 2} -> NRG
+    if enum_type_name == "int" or enum_type_name == "NoneType":
+        enum_field_names = list(enum_dict.keys())[:]
+        enum_type_name = "".join(name[0] for name in enum_field_names)
 
-    if enum_type_name == "int" or enum_type_name is None:
-        field_descriptor.type_name = ProtobufColor.DESCRIPTOR.full_name
-    elif enum_type_name not in [enum_type.name for enum_type in file_descriptor.enum_type]:
+    if enum_type_name not in [enum_type.name for enum_type in file_descriptor.enum_type]:
         enum_descriptor = file_descriptor.enum_type.add()
         enum_descriptor.name = enum_type_name
         for name, number in enum_dict.items():
             enum_value_descriptor = enum_descriptor.value.add()
-            enum_value_descriptor.name = name
+            enum_value_descriptor.name = f"{enum_type_name}_{name}"
             enum_value_descriptor.number = number
-        field_descriptor.type_name = enum_descriptor.name
-    else:
-        field_descriptor.type_name = enum_type_name
+    field_descriptor.type_name = enum_type_name
 
 
 def _get_enum_type(parameter_metadata: ParameterMetadata) -> type:
```

Comment:
decoder.py imports _get_enum_type, so you should remove the underscore to show that this is part of the public interface of the serialization_descriptors submodule (which is itself still private)